open-sphere-plugins/ogc-server/src/main/java/io/opensphere/server/control/AbstractServerSourceController.java

```diff
@@ -331,9 +349,69 @@ protected void updateSource(IDataSource source)
         finally
         {
             myConfigLock.writeLock().unlock();
+            if (myReloadListenerMap.containsKey(source))
+            {
+                EventListener<AbstractSingleStateEvent> listener = myReloadListenerMap.get(source);
+                myReloadListenerMap.remove(source);
+                listener.notify(new AbstractSingleStateEvent()
```

Comment:
Check for null on listener.

Reply:
I think the change I've made should be suitable.